statistics/testdata/stats_suite_out.json

```diff
@@ -7,6 +7,417 @@
+        "└─TableRowIDScan_9(Probe) 1.00 cop[tikv] table:tdouble keep order:false"
+      ],
+      [
+        "IndexLookUp_10 1.00 root  ",
+        "├─IndexRangeScan_8(Build) 1.00 cop[tikv] table:tdouble, index:multi(b, c) range:[8,8], keep order:false",
+        "└─TableRowIDScan_9(Probe) 1.00 cop[tikv] table:tdouble keep order:false"
+      ],
+      [
+        "IndexLookUp_10 1.00 root  ",
+        "├─IndexRangeScan_8(Build) 1.00 cop[tikv] table:tdecimal, index:multi(b, c) range:[1.00000000000000000000,1.00000000000000000000], keep order:false",
+        "└─TableRowIDScan_9(Probe) 1.00 cop[tikv] table:tdecimal keep order:false"
+      ],
+      [
+        "IndexLookUp_10 1.00 root  ",
+        "├─IndexRangeScan_8(Build) 1.00 cop[tikv] table:tdecimal, index:multi(b, c) range:[4.00000000000000000000,4.00000000000000000000], keep order:false",
+        "└─TableRowIDScan_9(Probe) 1.00 cop[tikv] table:tdecimal keep order:false"
+      ],
+      [
+        "IndexLookUp_10 1.00 root  ",
+        "├─IndexRangeScan_8(Build) 1.00 cop[tikv] table:tdecimal, index:multi(b, c) range:[8.00000000000000000000,8.00000000000000000000], keep order:false",
+        "└─TableRowIDScan_9(Probe) 1.00 cop[tikv] table:tdecimal keep order:false"
+      ],
+      [
+        "TableReader_7 1.00 root  data:Selection_6",
+        "└─Selection_6 1.00 cop[tikv]  eq(test.tstring.b, \"1\")",
+        "  └─TableFullScan_5 8.00 cop[tikv] table:tstring keep order:false"
+      ],
+      [
+        "TableReader_7 1.00 root  data:Selection_6",
+        "└─Selection_6 1.00 cop[tikv]  eq(test.tstring.b, \"4\")",
+        "  └─TableFullScan_5 8.00 cop[tikv] table:tstring keep order:false"
+      ],
+      [
+        "TableReader_7 1.00 root  data:Selection_6",
+        "└─Selection_6 1.00 cop[tikv]  eq(test.tstring.b, \"8\")",
+        "  └─TableFullScan_5 8.00 cop[tikv] table:tstring keep order:false"
+      ],
+      [
+        "TableReader_7 1.00 root  data:Selection_6",
+        "└─Selection_6 1.00 cop[tikv]  eq(test.tdatetime.b, 2001-01-01 00:00:00.000000)",
+        "  └─TableFullScan_5 4.00 cop[tikv] table:tdatetime keep order:false"
+      ],
+      [
+        "TableReader_7 1.00 root  data:Selection_6",
+        "└─Selection_6 1.00 cop[tikv]  eq(test.tdatetime.b, 2001-01-02 00:00:00.000000)",
+        "  └─TableFullScan_5 4.00 cop[tikv] table:tdatetime keep order:false"
+      ],
+      [
+        "TableReader_7 1.00 root  data:Selection_6",
+        "└─Selection_6 1.00 cop[tikv]  eq(test.tdatetime.b, 2001-01-04 00:00:00.000000)",
+        "  └─TableFullScan_5 4.00 cop[tikv] table:tdatetime keep order:false"
+      ],
+      [
+        "TableReader_6 4.00 root  data:TableRangeScan_5",
+        "└─TableRangeScan_5 4.00 cop[tikv] table:ct1 range:[\"1\",\"4\"], keep order:false"
+      ],
+      [
+        "TableReader_6 2.00 root  data:TableRangeScan_5",
+        "└─TableRangeScan_5 2.00 cop[tikv] table:ct1 range:[\"4\",\"6\"], keep order:false"
```

Comment:
This should be `3.00`.

Reply:
It is calculated in `Hist.BetweenRowCount`: 2 = lessCountRight - lessCountLeft = lessCount('7') - lessCount('4');

Since the first bucket's boundary is '4', lessCount('4') = bucket[0].count - bucket[0].repeat = 3;

lessCount('7') = bucket[0].count + bucket[1].fraction('7')*(bucket[1].count - bucket[0].count - bucket[1].repeat) = 4 + bucket[1].fraction('7')*3;
Now fraction calculation on string type is not accurate, but I think this error is acceptable.